### PR TITLE
[Feature](TrinoConnector-Catalog) Be supports trino-connector catalog

### PR DESCRIPTION
**Pull request by BePPPower**
## Proposed changes

Issue Number:  #31719 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BePPPower** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/trinoconnector/TrinoConnectorPluginLoader.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                typeRegistry = new TypeRegistry(typeOperators, featuresConfig);
+                ServerPluginsProviderConfig serverPluginsProviderConfig = new ServerPluginsProviderConfig()
+                        .setInstalledPluginsDir(new File(Config.trino_connector_plugin_dir));
+                ServerPluginsProvider serverPluginsProvider = new ServerPluginsProvider(serverPluginsProviderConfig,
+                        MoreExecutors.directExecutor());
+                trinoConnectorPluginManager = new TrinoConnectorPluginManager(serverPluginsProvider,
+                        typeRegistry, handleResolver);
+                trinoConnectorPluginManager.loadPlugins();
+            } catch (Exception e) {
+                LOG.warn("Exception when load trino-connector plugins from  " + Config.trino_connector_plugin_dir);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/trinoconnector/TrinoConnectorPluginLoader.java:56]: remove one of these 2 logs

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by kaka11chen — APPROVED** [review]
LGTM

**Pull request merged**